src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -1355,6 +1357,66 @@ s! {
         pub strchange_instrms: u16,
         pub strchange_outstrms: u16,
     }
+
+    pub struct filedesc {
+        pub fd_files: *mut fdescenttbl,
+        pub fd_map: *mut c_void,
+        pub fd_freefile: c_int,
+        pub fd_refcnt: c_int,
+        pub fd_holdcnt: c_int,
+        pub fd_sx: sx,
+        pub fd_kqlist: kqlist,
+        pub fd_holdleaderscount: c_int,
+        pub fd_holdleaderswakeup: c_int,
+    }
+
+    pub struct fdescenttbl {
+        pub fdt_nfiles: c_int,
+        pub fdt_ofiles: [filedescent; 0],
+    }
+
+    pub struct filedescent {
+        // This is normally `struct file`.
+        pub fde_file: *mut c_void,
```

Comment:
Are you defining it as "c_void" instead of `struct file` just so you won't have to add a definition for the latter, because you don't need it?  In that case, I suggest you make the field private.  That way it will be easier to change the field later.

Reply:
Yes, gonna do that.